### ISSUE TITLE
(docs) Field(min/max_length) should also refer to lists

### ISSUE THREAD

**Issue by straygar** [pydantic/pydantic#8771]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

The current docstring for the parameter is misleading (since `max_items` was removed in v1):
```
min_length: Minimum length for strings.
max_length: Maximum length for strings.
```

Maybe we should reword this to:
```
min_length: Minimum length for strings and lists.
max_length: Maximum length for strings and lists.
```

This was probably missed during the [v1 -> v2 migration](https://docs.pydantic.dev/latest/migration/#changes-to-pydanticfield)

### Python, Pydantic & OS Version

```Text
-
```

**Comment by sydney-runkle** [comment]
@straygar,

You're correct, thanks for catching this! Could you please open a PR with a fix? I'd be happy to review your changes.

**Comment by straygar** [comment]
Sure! I guess this also applies to sets and bytes? What'd be a good generic text? Do we even want to specify types here?

**Comment by sydney-runkle** [comment]
@straygar you can use "iterable" as a catch all term here. Thanks!

**Comment by jacob-petterle-at-benlabs** [comment]
Yes! @straygar `iterable` is more inclusive as then it is implied it works for ANY iterable (`tuples`, `dicts`, etc.)
